stdlib/dynArray.ml

```diff
@@ -0,0 +1,130 @@
+let baseLen = 50;;
+let multiplier = 2;;
+
+type 'a cell =
+  | Nil
+  | Cell of 'a
+;;
+
+type 'a dynArray = {
+    mutable arr: 'a cell array;
+    mutable len: int;
+  }
+;;
+
+exception OutOfBounds;;
+
+let isOutBounds a n =
+  n < 0 && n >= a.len
+;;
+
+let create () = {
+    arr = Array.make baseLen Nil;
+    len = 0;
+  }
+;;
+
+let make n x = {
+    arr = Array.make n (Cell x);
+    len = n;
+  }
+;;
+
+let extend a =
+  let len = Array.length a.arr in
+  let tmp = Array.make (len * multiplier) Nil in
+  for i = 0 to (a.len - 1) do
+    tmp.(i) <- a.arr.(i)
+  done;
+  a.arr <- tmp
+;;
+
+let append a x =
+  if a.len = (Array.length a.arr) then extend a;
+  a.arr.(a.len) <- Cell x;
+  a.len <- a.len + 1
+;;
+
+let get a n =
+  if isOutBounds a n then raise OutOfBounds;
+  match a.arr.(n) with
+  | Nil -> raise OutOfBounds
+  | Cell x -> x
+;;
+
+let set a n x =
+  if isOutBounds a n then raise OutOfBounds;
+  match a.arr.(n) with
+  | Nil -> raise OutOfBounds
+  | Cell _ -> a.arr.(n) <- Cell x
+;;
+
+let remove a n =
+  if isOutBounds a n then raise OutOfBounds;
+  for i = n to a.len - 2 do
+    a.arr.(i) <- a.arr.(i + 1)
+  done;
```

Comment:
This is the use case for `Array.blit` and anyway it is dubious if the interface should expose such a O(n) function.